jobsdb/jobsdb.go

```diff
@@ -1615,26 +1625,55 @@ func (jd *HandleT) invalidateCache(ds dataSetT) {
 	}
 }
 
-//Rename a dataset
-func (jd *HandleT) renameDS(ds dataSetT, allowMissing bool) {
+//mustRenameDS renames a dataset
+func (jd *HandleT) mustRenameDS(ds dataSetT) {
 	var sqlStatement string
 	var renamedJobStatusTable = fmt.Sprintf(`pre_drop_%s`, ds.JobStatusTable)
 	var renamedJobTable = fmt.Sprintf(`pre_drop_%s`, ds.JobTable)
-
-	if allowMissing {
-		sqlStatement = fmt.Sprintf(`ALTER TABLE IF EXISTS "%s" RENAME TO "%s"`, ds.JobStatusTable, renamedJobStatusTable)
-	} else {
+	err := jd.doInTransaction(func(tx *sql.Tx) error {
 		sqlStatement = fmt.Sprintf(`ALTER TABLE "%s" RENAME TO "%s"`, ds.JobStatusTable, renamedJobStatusTable)
-	}
-	_, err := jd.dbHandle.Exec(sqlStatement)
+		_, err := jd.dbHandle.Exec(sqlStatement)
+		if err != nil {
+			return err
```

Comment:
[minor] might be useful to return an error with a bit more context here like `return fmt.Errorf("could not rename status table: %w", err)`.

Reply:
yes, will do